File: ASPNetWebApi/MySuperApi/Models/MessageModels/ChatMessage.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MySuperApi.Models.MessageModels
{
    public class ChatMessage
    {
        public Guid Id { get; set; }
        public string? Content { get; set; }
        public DateTime SentAt { get; set; }

        [ForeignKey("AppUser")]
        public Guid SenderId { get; set; }
        [NotMapped]
        public AppUser? Sender { get; set; }
        
        [ForeignKey("Chat")]
        public Guid ChatId { get; set; }
        [NotMapped]
        public Chat? Chat { get; set; }
    }

}

File: MyPetAngularProject/src/app/models/chatMessage.ts
import {AppUser} from "./appUser";

export class ChatMessage {
  public id: string = '';
  public content: string = '';
  public sentAt: string = '';
  public senderId: string = '';
  public chatId: string = '';
  //Uses only in client
  public senderInformation: AppUser = new AppUser();
}
